File: DotNet/CodeSamplesForAssertion/DerivedClass.cs
namespace CodeSamplesForAssertion
{
    //Assert to verify the instance type of this class
    //Since this is a singleton, qualifies for another assert to two instances are same
    public class DerivedClass : IAnimal
    {
        private static readonly DerivedClass derivedClass_ = new DerivedClass();
        
        private DerivedClass(){

        }

        public static DerivedClass GetInstance()
        {
            return derivedClass_;
        }
        
        public void Move()
        {
            throw new System.NotImplementedException();
        }
    }

    public interface IAnimal
    {
        void Move();
    }
}

File: java/src/main/java/com/agile/CodeSamplesForAssertion/DerivedClass.java
package com.agile.CodeSamplesForAssertion;

public class DerivedClass implements Animal {

    private final static DerivedClass DERIVED_CLASS = new DerivedClass();

    //Assert to verify the instance type of this class
    //Since this is a singleton, qualifies for another assert to two instances are same
    private DerivedClass() {
    }

    public static DerivedClass getInstance() {
        return DERIVED_CLASS;
    }

    public void Move() {
        throw new UnsupportedOperationException();

    }

}